src/bun.js/base.zig

```diff
@@ -472,14 +472,33 @@ pub const ArrayBuffer = extern struct {
 
     const log = Output.scoped(.ArrayBuffer, false);
 
+    pub export fn Bun__ArrayBuffer__deallocate(bytes: ?[*]u8, byte_length: usize, allocatorPtr: ?*anyopaque, fntable: *const std.mem.Allocator.VTable) void {
+        const allocator = std.mem.Allocator{
+            // We use an "undefined" pointer for default_allocator
+            // So this value can genuinely be anything.
+            .ptr = allocatorPtr orelse undefined,
```

Comment:
should just change allocatorPtr to be a non-null pointer and let the caller pass in undefined memory, instead of having a runtime branch that willingly puts in undefined memory.